api/v1alpha1/etcdcluster_types.go

```diff
@@ -50,10 +53,56 @@ type ProviderConfig struct {
 	CertManagerCfg *ProviderCertManagerConfig `json:"certManagerCfg,omitempty"`
 }
 
+type AltNames struct {
+	// DNSNames accepts a string array, if empty programmatically populates the
+	// CommonName as DNSNames
```

Comment:
- I think it makes more sense to by default populate DNSNames as string something like `*.example.default.svc.cluster.local`, see https://github.com/etcd-io/etcd-operator/blob/1adf166c0efc0a7f593dc61d228f502ecda87772/internal/controller/utils.go#L86
- nit: Please ensure the comments' style is consistent as the existing ones (no need to to add "accepts a string" as the variable type already self-explains it), this comment applies to all fields, https://github.com/etcd-io/etcd-operator/blob/1adf166c0efc0a7f593dc61d228f502ecda87772/api/v1alpha1/etcdcluster_types.go#L33-L42
- All the fields in `CommonConfig` and `AltNames` are very common in certificate utilities/packages, please refer to how others (i.e. golang std lib) add comments for them.

Reply:
Updated the PR 👍